collectors/ebpf_process.plugin/ebpf_process.c

```diff
@@ -853,11 +867,164 @@ static int load_collector_file(char *path) {
     return 0;
 }
 
-int main(int argc, char **argv)
+static inline void ebpf_disable_apps() {
+    int i ;
+    for (i = 0 ;ebpf_modules[i].thread_name ; i++ ) {
+        ebpf_modules[i].apps_charts = 0;
+    }
+}
+
+static inline void ebpf_enable_specific_chart(struct  ebpf_module *em, int disable_apps) {
+    em->enabled = 1;
+    if (!disable_apps) {
+        em->apps_charts = 1;
+    }
+    em->global_charts = 1;
+}
+
+static inline void ebpf_enable_all_charts(int apps) {
+    int i ;
+    for (i = 0 ; ebpf_modules[i].thread_name ; i++ ) {
+        ebpf_enable_specific_chart(&ebpf_modules[i], apps);
+    }
+}
+
+static inline void ebpf_enable_chart(int enable, int disable_apps) {
+    int i ;
+    for (i = 0 ; ebpf_modules[i].thread_name ; i++ ) {
+        if (i == enable) {
+            ebpf_enable_specific_chart(&ebpf_modules[i], disable_apps);
+            break;
+        }
+    }
+}
+
+static inline void ebpf_set_thread_mode(int mode) {
+    int i ;
+    for (i = 0 ; ebpf_modules[i].thread_name ; i++ ) {
+        ebpf_modules[i].mode = mode;
+    }
+}
+
+void ebpf_print_help() {
+    const time_t t = time(NULL);
+    struct tm *ct = localtime(&t);
+
+    fprintf(stderr,
+            "\n"
+            " Netdata ebpf.plugin %s\n"
+            " Copyright (C) 2016-%d Costa Tsaousis <costa@tsaousis.gr>\n"
+            " Released under GNU General Public License v3 or later.\n"
+            " All rights reserved.\n"
+            "\n"
+            " This program is a data collector plugin for netdata.\n"
+            "\n"
+            " Available command line options:\n"
+            "\n"
+            " SECONDS              set the data collection frequency.\n"
+            "\n"
+            " help or -h or -H      show this help.\n"
+            "\n"
+            " version or -v or -V   show software version.\n"
+            "\n"
+            " global or -g or -G    disable charts per application.\n"
+            "\n"
+            " all or -a or -A       Enable all chart groups (global and apps), unless -g is also given.\n"
+            "\n"
+            " net or -n or -N       Enable network viewer charts.\n"
+            "\n"
+            " process or -p or -P   Enable charts related to process run time.\n"
+            "\n"
+            " return or -r or -R    Run the collector in return mode.\n"
+            "\n"
+            , VERSION
+            , (ct->tm_year >= 116)?ct->tm_year + 1900: 2020
+    );
+}
+
+static void parse_args(int argc, char **argv)
 {
-    (void)argc;
-    (void)argv;
+    int i, freq = 0;
+    int enabled = 0;
+    int disable_apps = 0;
+    for (i = 1; i < argc ; i++) {
+        char *w = argv[i];
+        if (!freq) {
+            int n = (int)str2l(w);
+            if(n > 0) {
+                freq = n;
+                continue;
+            }
+        }
 
+        if (strcmp("help", w) == 0 || strcmp("-help", w) == 0 || strcmp("--help", w) == 0 || strcmp("-h", w) == 0 || strcmp("-H", w) == 0) {
```

Comment:
Maybe do something like this?
https://github.com/netdata/netdata/blob/166958349599708ae34397a8407aa35982f8c81c/daemon/main.c#L912

Reply:
I do not have any objection to use `getopt` or `getopt_long`, I used this algorithm, because I imagined we had a different pattern for plugins:

https://github.com/netdata/netdata/blob/166958349599708ae34397a8407aa35982f8c81c/collectors/apps.plugin/apps_plugin.c#L3855

https://github.com/netdata/netdata/blob/166958349599708ae34397a8407aa35982f8c81c/collectors/cups.plugin/cups_plugin.c#L117

Reply:
We can keep this as is for now, but we should keep in mind that the ebpf plugin will grow significantly and at some point we should refactor this anyway.

Reply:
I convert to `getopt_long`, so we do not need to change in the future.